### ISSUE TITLE
info["means2D"] has absgrad but no grad

### ISSUE THREAD

**Issue by Metro1998** [nerfstudio-project/gsplat#390]
First of all, thank you very much for open-sourcing such an excellent CUDA library. The issue I'm facing is that when I set absgrad to true, info[means2D].absgrad exists after loss.backward(). However, when I set absgrad to false, after loss.backward(), I can't access info[means2D].grad, and it is None.

**Comment by HamiltonHuaji** [comment]
Calling info[means2D].retain_grad() manually after rasterization before loss.backward() works for me.

**Comment by Metro1998** [comment]
Fixed with info[means2D].retain_grad()
